resources/inputs/register-form.tsx

```diff
@@ -0,0 +1,129 @@
+        data.password,
+      );
+      if (response != null) {
+        messageHandler.handleSuccess(t("Registration successful"));
+        onRegisterSuccess(response.user, response.token as string);
+      }
+    } catch (error) {
+      if (error instanceof Error) {
+        messageHandler.handleError(t(error.message));
+      } else {
+        messageHandler.handleError(t("An unexpected error occurred"));
+      }
+    }
+  };
+
+  const handleOAuthClick = async (provider: Provider) => {
+    try {
+      const response = await AuthService.oauth(provider);
+      if (response && response.url) window.location.href = response.url;
+    } catch (error) {
+      if (error instanceof Error) {
+        messageHandler.handleError(t(error.message));
+      } else {
+        messageHandler.handleError(
+          t("An unexpected error occurred during OAuth login"),
+        );
+      }
+    }
+  };
+
+  return (
+    <form
+      onSubmit={handleSubmit(onSubmit)}
+      className={form_style.formContainer}
+    >
+      <div className={form_style.formInput}>
+        <input
+          {...register("email", { required: "Email is required" })}
+          type="email"
+          placeholder={t("Email")}
+          className={`${form_style.formControl} ${form_style.inputEmailIcon}`}
+        />
+        {errors.email && (
+          <span className={form_style.errorMessage}>
+            {t(errors.email.message)}
+          </span>
+        )}
+      </div>
+      <div className={form_style.formInput}>
+        <input
+          {...register("password", { required: "Password is required" })}
+          type="password"
+          placeholder={t("Password")}
+          className={`${form_style.formControl} ${form_style.inputPasswordIcon}`}
+        />
+        {errors.password && (
+          <span className={form_style.errorMessage}>
+            {t(errors.password.message)}
+          </span>
+        )}
+      </div>
+      <div className={form_style.buttonWrapper}>
+        <button type="submit" className={form_style.submitButton}>
+          {t("Register")}
+        </button>
+      </div>
+      <div className={form_style.oauth}>
+        <div className={form_style.oauthTextContainer}>
+          <p className={form_style.oauthText}>
+            <strong>{t("Register")}</strong> {t("with others")}:
+          </p>
+        </div>
+        <button
+          type="button"
+          onClick={() => handleOAuthClick("google")}
+          className={`${form_style.oauthButton} ${form_style.googleOAuthButton}`}
+        >
+          <p>
+            {" "}
+            {t("Register with")} <strong> google</strong>{" "}
+          </p>
+        </button>
+        <button
+          type="button"
+          onClick={() => handleOAuthClick("facebook")}
+          className={`${form_style.oauthButton} ${form_style.facebookOAuthButton}`}
+        >
+          <p>
+            {" "}
+            {t("Register with")} <strong> facebook</strong>{" "}
```

Comment:
Facebook and google shall be translated.

For example in Chinese the name change :D (just in case)